language/move-lang/tests/functional/sliding_nonce/test.move

```diff
@@ -0,0 +1,217 @@
+//! account: bob, 100000000, 0, unhosted
+//! account: alice, 100000000, 0, unhosted
+
+// ****
+// Account setup - bob is account with nonce resource and alice is a regular account
+// ****
+
+//! new-transaction
+//! sender: bob
+script {
+    use 0x0::SlidingNonce;
+    // Unset testnet
+    fun main() {
+        SlidingNonce::create_nonce_resource_for_user()
+    }
+}
+// check: EXECUTED
+
+// ****
+// Basic tests - when nonce == 0 check always pass, even if user don't have nonce resource
+// ****
+
+//! new-transaction
+//! sender: alice
+script {
+    use 0x0::SlidingNonce;
+    // Unset testnet
+    fun main() {
+        SlidingNonce::accept_nonce(0)
+    }
+}
+// check: EXECUTED
+
+//! new-transaction
+//! sender: alice
+script {
+    use 0x0::SlidingNonce;
+    // Unset testnet
+    fun main() {
+        SlidingNonce::accept_nonce(0)
+    }
+}
+// check: EXECUTED
+
+//! new-transaction
+//! sender: bob
+script {
+    use 0x0::SlidingNonce;
+    // Unset testnet
+    fun main() {
+        SlidingNonce::accept_nonce(0)
```

Comment:
Might be easier to read/maintain if we combine all of the successful `bob` tests into one  (this is what we tend to do elsewhere in the tests):

```
 fun main() {
     SlidingNonce::create_nonce_resource_for_user();
     SlidingNonce::accept_nonce(0);
     SlidingNonce::accept_nonce(0); // duplicates ok
     SlidingNonce.accept_nonce(1);  
     SlidingNonce::accept_nonce(1000 + 127);
     SlidingNonce::accept_nonce(1000);
     SlidingNonce::accept_nonce(2000 + 128);
     SlidingNonce::accept_nonce(3000 + 128);
     SlidingNonce::accept_nonce(3000 + 127)
}
```

Reply:
Also, one  test with a loop where you try all the values between `N` and 128 might be useful...

Reply:
So problem here is that every time something fails it aborts the whole thing. How do I test that?
How do I do this check in the same fn main?
```
     SlidingNonce::accept_nonce(2000 + 128); // accepted
     SlidingNonce::accept_nonce(2000); // failed
```

Reply:
Ah, that you cannot do. I was suggesting combining all of  the success tests into one fn (since most of these are success cases). 

For the failure ones, each needs their own separate fn, but in this case that seems reasonable because the code to build up the state to the failure case is so minimal. But ok with any test organization, just a suggestion!

Reply:
I guess one more thing that you could do is factor out `accept_nonce(u64): boolean` a, call it from some new fn like `advance_nonce()`, and check check `accept_nonce() == false` instead of failure tests